yolox/data/datasets/datasets_wrapper.py

```diff
@@ -112,3 +122,143 @@ def wrapper(self, index):
             return ret_val
 
         return wrapper
+
+
+class CacheDataset(Dataset, metaclass=ABCMeta):
+    def __init__(
+        self,
+        input_dimension,
+        num_imgs=None,
+        data_dir=None,
+        cache_dir_name=None,
+        path_filename=None,
+        cache=False,
+        cache_type="ram",
+    ):
+        super().__init__(input_dimension)
+        self.cache = cache
+        self.cache_type = cache_type
+
+        if self.cache and self.cache_type == "disk":
+            self.cache_dir = os.path.join(data_dir, cache_dir_name)
+            self.path_filename = path_filename
+
+        if self.cache and self.cache_type == "ram":
+            self.imgs = None
+
+        if self.cache:
+            self.cache_images(
+                num_imgs=num_imgs,
+                data_dir=data_dir,
+                cache_dir_name=cache_dir_name,
+                path_filename=path_filename,
+            )
+
+    def __del__(self):
+        if self.cache and self.cache_type == "ram":
+            del self.imgs
+
+    @abstractmethod
+    def read_img(self, index, use_cache=True):
+        raise NotImplementedError
+
+    @staticmethod
```

Comment:
why not moving `cache_read_img` outside this class?

Reply:
`import CacheDataset` will automatically import `cache_read_img` function, but if moving `cache_read_img` outside this class, we will use `from .datasets_wrapper import CacheDataset, cache_read_img`

Reply:
IMHO, the second one is better since `cache_read_img` is a seprate logic.